File: November2022/Java/RemoveDuplicatesFromSortedArray.java
// @saorav21994

class Solution {
    public int removeDuplicates(int[] nums) {
        int l = nums.length;
        if (l == 0 || l == 1) {
            return l;
        }
        int i = 0, j = 0;
        int count = 1;
        for (j = 1; j < l; j++) {
            if (nums[i] == nums[j]) {
                // Do nothing
            }
            else {
                i += 1;
                count += 1;
                nums[i] = nums[j];
            }
        }
        return count;
    }
}

File: November2022/C++/remove-duplicates-from-sorted-array.cpp
// @Sarang661
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        // 1 2 3 3 3 3 3 4 4 5 5 5
        int len = 1;
        int start = 0;
        int end = 0;
        
        for(int index = 0;index<nums.size();index++){
            end = index;
            int k = nums[end];
            
            if(k!=nums[start]){
                len++;
                nums[start+1] = k;
                start++;
            }
        }
        
        return len;
    }
};